### PR TITLE
fix: locale uses preferred language

### PR DESCRIPTION
**Pull request by levibostian**
Currently, the device locale that the SDK obtains is the current locale of the host app. This has limitations on it because if the user's device is set to Spanish but the host app does not support Spanish, the device locale that the SDK receives is not Spanish. 

We want to improve our SDK's code in regards to obtaining the device locale. We would prefer is we get the locale of the device even if the host app does not support it. That's what this PR is doing. 

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/GIT-WORKFLOW.md). 
- [ ] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request. 
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1). 
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request. A bot will *squash and merge* the pull request for you after the label is added.

### CODE REVIEW

File: Sources/Tracking/Util/DeviceInfo.swift [hunk 1]
```diff
@@ -66,8 +66,22 @@ public class CIODeviceInfo: DeviceInfo {
         SdkVersion.version
     }
 
+    // Requirements:
+    // 1. Use - instead of _
+    // 2. We prefer to get the OS language that the user has set. First try to return that. If that does not succeed
+    // we default to the language set for the host app. The language returned for that will only return languagues
+    // that the app supports. If OS set to "es" but app does not support Spanish, then "es" will not be returned.
```
Comment (jatinn) [inline Sources/Tracking/Util/DeviceInfo.swift:73]: `If OS set to "es" but app does not support Spanish, then "es" will not be returned.`

comment seems inconsistent with the code to me as looks like we will always prefer to return the os lang.

File: Sources/Tracking/Util/DeviceInfo.swift [hunk 2]
```diff
@@ -66,8 +66,22 @@ public class CIODeviceInfo: DeviceInfo {
...
     public var deviceLocale: String {
-        Locale.current.identifier.replacingOccurrences(of: "_", with: "-")
+        if let osSetLanguage = Locale.preferredLanguages.first {
+            let locale = Locale(identifier: osSetLanguage)
+
+            if let languageCode = locale.languageCode,
```
Comment (jatinn) [inline Sources/Tracking/Util/DeviceInfo.swift:78]: could this be simplified to
```
return locale.identifier.replacingOccurrences(of: "_", with: "-")
or
return osSetLanguage.replacingOccurrences(of: "_", with: "-")
```

Comment (jatinn) [inline Sources/Tracking/Util/DeviceInfo.swift:78]: could this be simplified to
```
return locale.identifier.replacingOccurrences(of: "_", with: "-")
or
return osSetLanguage.replacingOccurrences(of: "_", with: "-")
```

**Review by jatinn — APPROVED** [review]

File: Sources/Tracking/Util/DeviceInfo.swift (hunk 2 above)
Comment (levibostian) [reply Sources/Tracking/Util/DeviceInfo.swift:78]: duplicate. Closing.

Comment (levibostian) [reply Sources/Tracking/Util/DeviceInfo.swift:78]: Looking at the code, I don't currently see an easy way to do that. There are so many possibilities for `null` to happen that's why this code has 3 `if let` statements.

File: Sources/Tracking/Util/DeviceInfo.swift (hunk 1 above)
Comment (levibostian) [reply Sources/Tracking/Util/DeviceInfo.swift:73]: `if let osSetLanguage = Locale.preferredLanguages.first` has the possibility of returning `null` and the `if` statement being `false`. That's when we will *not* be returning the OS lang.

File: Sources/Tracking/Util/DeviceInfo.swift (hunk 2 above)
Comment (levibostian) [reply Sources/Tracking/Util/DeviceInfo.swift:78]: Looking at the code, I don't currently see an easy way to do that. There are so many possibilities for null to happen that's why this code has 3 `if let` statements.

**Pull request merged**